File: src/api/MintyPeterson.Counter.Api/Models/Responses/EntryListEntryResponse.cs
// <copyright file="EntryListEntryResponse.cs" company="Tom Cook">
// Copyright (c) Tom Cook. All rights reserved.
// </copyright>

namespace MintyPeterson.Counter.Api.Models.Responses
{
  /// <summary>
  /// Represents an entry in a <see cref="EntryListResponse"/>.
  /// </summary>
  public class EntryListEntryResponse
  {
    /// <summary>
    /// Gets or sets the entry identifier.
    /// </summary>
    public Guid EntryId { get; set; }

    /// <summary>
    /// Gets or sets the entry date.
    /// </summary>
    /// <remarks>Only the date part is retained.</remarks>
    public DateTime EntryDate { get; set; }

    /// <summary>
    /// Gets or sets the entry.
    /// </summary>
    public decimal Entry { get; set; }

    /// <summary>
    /// Gets or sets the notes.
    /// </summary>
    public string? Notes { get; set; }

    /// <summary>
    /// Gets or sets a value indicating whether the entry is an estimate or not.
    /// </summary>
    public bool IsEstimate { get; set; }
  }
}

File: src/flutter/lib/services/counter/responses/entry_list_entry_response.dart
import 'package:counter/services/counter/responses/entry_list_response.dart';
import 'package:json_annotation/json_annotation.dart';

part 'entry_list_entry_response.g.dart';

/// Represents an entry in a [EntryListResponse].
@JsonSerializable(createToJson: false)
class EntryListEntryResponse {

  final String entryId;
  final DateTime entryDate;
  final int entry;
  final String? notes;
  const EntryListEntryResponse({
    required this.entryId,
    required this.entryDate,
    required this.entry,
    required this.notes,
  });

  factory EntryListEntryResponse.fromJson(Map<String, dynamic> json) =>
    _$EntryListEntryResponseFromJson(json);
}